powertools-idempotency/src/main/java/software/amazon/lambda/powertools/idempotency/persistence/DataRecord.java

```diff
@@ -26,9 +26,30 @@
 public class DataRecord {
     private final String idempotencyKey;
     private final String status;
+
+    /**
+     * This field is control how long the result of the idempotent
+     * event is cached. It is stored in _seconds since epoch_.
+     *
+     * DynamoDB's TTL mechanism is used to remove the record once the
+     * expiry has been reached, and subsequent execution of the request
+     * will be permitted. The user must configure this on their table.
+     */
     private final long expiryTimestamp;
     private final String responseData;
     private final String payloadHash;
+
+    /**
+     * The in-progress field is set to the remaining lambda execution time
+     * when the record is created.
+     * This field is stored in _milliseconds since epoch_.
+     *
+     * This ensures that:
+     *
+     * 1/ other concurrently executing requests are blocked from starting
+     * 2/ if a lambda times out, subsequent requests will be allowed again, despite
+     *     the fact that the idempotency record is already in the table
+     */
```

Comment:
I could also rename this field completely - `inProgressExpiryTimestampMs`. It's not public. WDYT?

Reply:
I've changed my mind on this. Because the setter/getter are public (should they be?) it doesn't make much sense to rename the field alone.

Reply:
I've changed my mind on this. Because the setter/getter are public (should they be?) it doesn't make much sense to rename the field alone.

Reply:
let's keep them like this, it would be strange to specify the unit for one and not the other.